### DISCUSSION TITLE
supabase.auth.user() returns undefined

### DISCUSSION THREAD

**Discussion by khansherdil** [supabase/supabase#5272] [Questions] [answered]
Hi guys, I'm trying to query for a matching uid so I can manage user sessions in my application, however, when passing in the id from supabase.auth.users() I get a 400.

#### Error messages I get
next-dev.js?5321:32 Error: invalid input syntax for type uuid: "undefined"
400 status because I'm passing in an undefined I assume

```tsx
const getUser = async (id: any) => {
  let { data: users, error: FetchingUIDError } = await supabase
    .from<User>("users")
    .select()
    .eq("id", id)
    .single();

  if (FetchingUIDError) {
    throw new Error(FetchingUIDError.message);
  }

  if (!users) {
    throw new Error("User with that ID not found");
  }

  return users;
};

```

This is the React Query hook I use it in: 

```tsx
const useUser = () => {
  const user = supabase.auth.user(); //error here?
//if i console log user?.id I get undefined
  return useQuery("user", () => getUser(user?.id)); 
};

export default useUser;

```

If I pass in a uid from one of my users from the users table in my getUser() I get all the corresponding details. 
So the problem really lies with the user?.id being undefined. 
Would really appreciate some insight, been stuck at this for the past few hours 😅

**Comment by GaryAustin1** [comment, marked as answer]
If user is undefined, then you have no logged in user in the client yet.  Either a timing issue, or you're running the code on your server at this point and need to use session call.
Make sure you wait for signIn() completion before calling auth.user()

This is all auth.user() does:
```
/**
   * Inside a browser context, `user()` will return the user data, if there is a logged in user.
   *
   * For server-side management, you can get a user through `auth.api.getUserByCookie()`
   */
  user(): User | null {
    return this.currentUser
  }
```

**Reply by khansherdil** [reply]
Thank you. 
Looks like there was no logged in user after all. Missed some crucial login logic.

**Reply by Felix-Okeyo** [reply]
Hey I am having a similar error in my code when I try login in. I had deployed the app and was working well till I deleted a few users at Supabase. None is logged in and when I try login in using my Auth component which was working before, I am getting a 404 "invalid input syntax for type uuid: "undefined" "

**Comment by simonh1000** [comment]
I'm seeing this too. If I leave the site alone for a few days, then the first time I go back to it `supabase.auth.user` does not return a user. But if I then simply refresh the page, there is a user. Needless to say, a 1+ day iteration cycle is a pain to debug!

**Reply by GraysonQuartech** [reply]
@simonh1000  did u resolve this issue? facing it right now with signInWithAuth()  ... let me know!

**Reply by simonh1000** [reply]
I found a variety of issues in my code. I think the best I can offer is to make sure that you are redirecting back to `/auth/callback` (in SvelteKit) so that you are converting the code into a session

**Comment by chriscarrollsmith** [comment]
I have recently been experiencing this issue after upgrading to Next.js 14.1. It went away after I rolled back to Next.js 14.0.4. Not sure what's causing it. Maybe some change in middleware behavior.

**Reply by chriscarrollsmith** [reply]
Ultimately determined that the 'use server' directive in 'utils/supabase/server' was causing the error. The bug went away once I removed this directive. Puzzling, because we're creating a server client with createServerClient, so I would think this would run on the server. But maybe it's running partly on the server and partly on the client, since it has to interface with cookies?
